### PR TITLE
Change to use declarative types with @graphql-tools/schema

### PR DESCRIPTION
**Pull request by salma-bek**
The purpose of this PR is to use declarative types with @graphql-tools/schema instead of GraphQL’s reference implementation

### CODE REVIEW

File: javascript/MLE-Demo/book-schema.mjs [hunk 1]
```diff
@@ -13,151 +13,88 @@ See the License for the specific language governing permissions and
...
+    library: async (_, { research }) => {
+      return await getBooks(research);
+    }
+  },
+  VolumeInfo: {
+    authors: (volumeInfo) => volumeInfo.authors || [],
+  },
+  Book: {
+    stock: async (book) => {
+      const conn = oracledb.defaultConnection();
```
Comment (martin-bach-oracle) [inline javascript/MLE-Demo/book-schema.mjs:74]: please consider using `session.execute()` here.

File: javascript/MLE-Demo/book-schema.mjs [hunk 2]
```diff
@@ -13,151 +13,88 @@ See the License for the specific language governing permissions and
...
+        return result.rows[0].STOCK;
+      } else {
+        return 0;
+      }
+    }
+  },
+};
+
+const getBooks = async (SEARCH_URL) => {
   const conn = oracledb.defaultConnection();
```
Comment (martin-bach-oracle) [inline javascript/MLE-Demo/book-schema.mjs:86]: please consider using `session.execute()` instead

File: javascript/MLE-Demo/book-schema.mjs [hunk 3]
```diff
@@ -13,151 +13,88 @@ See the License for the specific language governing permissions and
...
-  begin
-     utl_http.set_wallet('file:/<absolute-path-to-your-wallet-directory>');
-  end;
+  await conn.execute(`
+    begin
+      utl_http.set_wallet('file:/<absolute-path-to-your-wallet-directory>');
+    end;
   `);
-  
   const response = await fetch(`${BASE_URL}${SEARCH_URL}`, { credentials: "include" });
```
Comment (martin-bach-oracle) [inline javascript/MLE-Demo/book-schema.mjs:92]: beginning with Oracle Database 23.3 it should no longer be necessary to use `createntials: include`

**Comment by martin-bach-oracle** [comment]
In addition to the comments in the code, please add instructions how to use rollup to generate a bundle of `graphql-tools` and your code.

**Review by martin-bach-oracle — CHANGES_REQUESTED** [review]
Kindly address the outstanding comments and resubmit the code for review

**Comment by gvenzl** [comment]
Closed as per @martin-bach-oracle.

**Pull request closed without merging**